### PR TITLE
AFXM-167: Main Navigation

### PR DESCRIPTION
**Pull request by ajit-verma-altudo**
- NavigationMenu Component
- useIsMediumScreen Hook
- HamburgerButton Component
- DesktopNavigation Component
- MobileNavigation Component

### CODE REVIEW

File: src/aflglobal/src/core/molecules/DesktopNavigationMenu/DesktopNavigationMenu.module.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@import '/src/assets/sass/abstracts/vars';
+
+.mainNavigation {
+  display: flex;
+  flex: auto;
+  justify-content: flex-end;
+  padding: 0 $space-25;
+  .navigationItems {
+    padding: 0;
+    margin: 0;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/DesktopNavigationMenu/DesktopNavigationMenu.module.scss:10]: use variable.

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
+@import '/src/assets/sass/abstracts/vars';
+
+.navigationItems {
+  position: fixed;
+  background: $bg-secondary-menu;
+  z-index: 99;
+  top: $space-55;
+  bottom: 0;
+  right: -100%;
+  left: 100%;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:10]: use variable.

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
+@import '/src/assets/sass/abstracts/vars';
+
+.navigationItems {
+  position: fixed;
+  background: $bg-secondary-menu;
+  z-index: 99;
+  top: $space-55;
+  bottom: 0;
+  right: -100%;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:9]: use variable. and do not use -100%

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  left: 100%;
+  transition: 0.2s ease-in-out;
+  padding: 0;
+  margin: 0;
+  &.menuOpen {
+    right: 0;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:15]: use variable.

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    left: 0;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:16]: use variable.

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          display: inline-block;
+          border: $border-2 $text-blue;
+          transform: rotate($angle-45) translateY(-50%);
+          border-left-color: transparent;
+          border-bottom-color: transparent;
+          width: $size-12;
+          height: $size-12;
+          position: absolute;
+          right: $space-10;
+          top: 50%;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:40]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+@import '/src/assets/sass/abstracts/vars';
+@import '/src/assets/sass/abstracts/mixins';
+
+.newsCard {
+  position: relative;
+  border-radius: $border-radius-26;
+  width: 26rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:7]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  height: 29rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:8]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  background-repeat: no-repeat;
+  background-size: cover;
+  background-position: top;
+  flex: auto;
+
+  @include respond-to(lg) {
+    width: 20rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:15]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    height: 21rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:16]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  @include respond-to(md) {
+    width: 13rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:20]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    height: 14rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:21]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  .titleItemDetail {
+    background-color: #e1e1e1;
+    border-radius: 0 0 $border-radius-26 $border-radius-26;
+    padding: 20px 15px;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:27]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    margin-top: auto;
+    position: absolute;
+    bottom: 0;
+    width: 100%;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:31]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    .title {
+      font-size: 2.17rem;
+      font-weight: 600;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:34]: use variable. and do not use width use logical property.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+      color: #323232;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:35]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+      line-height: 23px;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:36]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+      margin: 0;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:37]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+      display: -webkit-box;
+      -webkit-line-clamp: 2;
+      -webkit-box-orient: vertical;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
+    .date {
+      font-size: 1.2rem;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:45]: use variable.

File: src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+      color: #8b8b8b;
+      font-weight: bold;
+      letter-spacing: 1.2px;
```
Comment (rgaur2312) [inline src/aflglobal/src/core/molecules/NewsCard/NewsCard.module.scss:48]: use variable.

**Review by rgaur2312 — COMMENTED** [review]
@ajit-verma-altudo  : kindly resolve feedback

File: src/aflglobal/src/assets/sass/abstracts/vars/_borders.scss [hunk 21]
```diff
@@ -2,6 +2,8 @@ $border-1: 1px solid;
 $border-2: 2px solid;
 $border-3: 4px solid;
 $border-4: 8px solid;
+$border-5: 5px solid;
```
[the remark this answers is not in the document]
Comment (ajit-verma-altudo) [reply src/aflglobal/src/assets/sass/abstracts/vars/_borders.scss:5]: I have created this file for border-widths basically, so I can remove border_style from here.

File: src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss (hunk 3 above)
Comment (ajit-verma-altudo) [reply src/aflglobal/src/core/molecules/MobileNavigationMenu/MobileNavigationMenu.module.scss:9]: I have created variables. and We need to do this (negative %) for the transition. if there any other alternative, please suggest.

**Pull request merged**